src/pages/en/guides/rss.mdx

```diff
@@ -34,115 +34,156 @@ First, install `@astrojs/rss` using your preferred package manager:
 
-Say your blog posts are stored under the `src/pages/blog/` directory. You can generate an RSS feed like so:
+For example, say you have a `blog` collection under `src/content/blog/`. You can retrieve this collection and generate an RSS feed like so:
 
-```js
-// src/pages/rss.xml.js
+```js title="src/pages/rss.xml.js" "items:" "const blog = await getCollection('blog');"
 import rss from '@astrojs/rss';
+import { getCollection } from 'astro:content';
+
+export async function get(context) {
+  const blog = await getCollection('blog');
+  return rss({
+    title: 'Buzz’s Blog',
+    description: 'A humble Astronaut’s guide to the stars',
+    site: context.site,
+    items: blog.map((post) => ({
+      title: post.data.title,
+      pubDate: post.data.pubDate,
+      description: post.data.description,
+      customData: post.data.customData,
+      // Compute RSS link from post `slug`
+      // This example assumes all posts are rendered as `/blog/[slug]` routes
+      link: `/blog/${post.slug}/`,
+    })),
+  });
+}
+```
 
-export const get = () => rss({
-  title: 'Buzz’s Blog',
-  description: 'A humble Astronaut’s guide to the stars',
-  site: import.meta.env.SITE,
-  items: import.meta.glob('./blog/**/*.md'),
+:::tip
+The `link` property of each feed item can be computed from the content entry `slug`. See [our guide on generating pages from content collections](/en/guides/content-collections/#generating-pages-from-content-collections) to see how slugs correspond to routes.
+:::
+
+`@astrojs/rss` also exposes an `rssSchema` for use with collection schemas. This ensures your frontmatter contains all properties expected by an RSS feed item. You can apply this schema to your content config like so:
+
+```js title="src/content/config.ts" "rssSchema"
+import { rssSchema } from '@astrojs/rss';
+
+const blog = defineCollection({
+  schema: rssSchema,
 });
+
+export const collections = { blog };
 ```
 
-See [Vite's glob import documentation](https://vitejs.dev/guide/features.html#glob-import) for more on this import syntax.
+### Using glob imports
 
-### 2. List of RSS feed objects
+You may generate an RSS feed from documents in `src/pages/`. Since `Astro.glob` is not available from endpoint routes, we've included a `pagesGlobToRssItems()` helper. This accepts an `import.meta.glob` result ([see Vite documentation](https://vitejs.dev/guide/features.html#glob-import)) and outputs an array of valid RSS feed items.
 
-We recommend this option for `.md` files outside of the `pages` directory. This is common when generating routes [via `getStaticPaths`](/en/reference/api-reference/#getstaticpaths).
+You can apply `pagesGlobToRssItems()` like so. This function assumes that you are globbing for items inside `src/pages/`, and all necessary feed properties are present in each document's frontmatter.
```

Comment:
```suggestion
This function assumes, but does not verify, that all necessary feed properties are present in each document's frontmatter. If you encounter errors, verify each page frontmatter manually.
```
Again, I think the examples are good enough to speak for themselves. So, we can use this as an opportunity to emphasize that this is DIY when it comes to making sure you have the required frontmatter properties. So something like this is maybe more helpful as the user-facing content?